96 coins amount abbreviations

HellRemus opened the pull request:
**Features**
This adds #96 so that players can use K, M or B to symbolise 1000, 1000 000 and 1000 000 000 respectively. Case-insensitive. Works for all commands that inputs an amount. 

Examples:
`/bal set 1.2K Steve`
`/bal send 99B Dinnerbone`

**Technical changes**
* Dependency updates:
Update PaperAPI to `1.21.8-R0.1-SNAPSHOT`
Update HikariCP to `7.0.1`
Make shure JUnit, Jackson and Maven are the latest versions
* Translations:
Added `balcommand.abbreviation.invalid`
* BalanceCommand.java:
Methods now accept abbreviations by multiplication.
* Error handling:
Invalid number formats trigger existing "Invalid number!" message
Invalid abbreviations trigger new specific error message explaining valid options
Case-insensitive parsing (both uppercase and lowercase work)

**Limitations**
* Shop creation does NOT support abbreviations.

HellRemus committed:
Update dependencies and version to 1.2-SNAPSHOT-3-MC1.21.8, including Paper API, HikariCP, Jackson, and JUnit.

HellRemus committed:
Ensure tables are fully created before upgrades to prevent race conditions. Update foreign key removal in upgrade queries for Transactions table.

HellRemus committed:
Update .gitignore to include .cursor directory for user-specific files.

HellRemus committed:
Introduce AbbreviationUtil for parsing monetary values with abbreviations. Update translations for new error messages related to invalid abbreviations and numbers.

HellRemus committed:
Merge branch 'next-update-bug-fixes'

HellRemus committed:
Finish merging next-update-bug-fixes and apply Brigadier changes

HellRemus committed:
Add detailed Javadoc comments for BalanceCommand, BankCommand, and QuickeconomyCommand classes

HellRemus committed:
Refactor BalanceCommand to handle UUID retrieval exceptions gracefully and update TypeChecker to improve UUID formatting logic.

affekri commented on src/main/java/net/derfla/quickeconomy/command/BankCommand.java:
Extensive Javadoc is removed?

affekri commented on src/main/java/net/derfla/quickeconomy/command/BalanceCommand.java:
Extensive Javadocs are removed?

affekri commented on src/main/java/net/derfla/quickeconomy/command/BalanceCommand.java:
Maybe a custom coin amount argument could fit better here, instead of using string. This will handle the error checking in one place instead of having one check in every subcommand.

affekri commented on src/main/java/net/derfla/quickeconomy/command/BalanceCommand.java:
'/bal list' should be handled in the 'runListLogic' method.

affekri commented on src/main/java/net/derfla/quickeconomy/command/BalanceCommand.java:
com.mojang.brigadier.Command.SINGLE_SUCCESS; is overly verbose.
Command.SINGLE_SUCCESS; should suffice.

HellRemus committed:
Make BalanceCommand to use MoneyArgument for monetary inputs, simplified money parsing logic and better execution flow.

HellRemus committed:
Add MoneyArgument class for custom monetary input parsing in commands

Implements a Brigadier argument type that allows for parsing monetary amounts with optional abbreviation suffixes (K, M, B, T). Includes error handling and suggestions for tab-completion.

HellRemus committed:
Improved parsing logic and Javadocs. Added support for trillion (T) suffix and refined error handling.

HellRemus committed:
Update translations to include 'T' as a valid abbreviation for monetary values in English and Swedish properties files.
